src/v/cluster/rm_stm.cc

```diff
@@ -427,7 +427,6 @@ ss::future<checked<model::term_id, tx_errc>> rm_stm::do_begin_tx(
     // strictly after all records are written it means that it
     // won't be retrying old writes and we may reset the seq cache
     _log_state.seq_table.erase(pid);
-    _mem_state.inflight[pid] = 0;
 
     co_return synced_term;
 }
```

Comment:
> With this PR rm_stm become more robust and tolerates reboots 

It only tolerates graceful transfers, no?

Reply:
actually it tolerates non graceful transfers too